zip-tor-txn-relay-network.rst

```diff
@@ -0,0 +1,132 @@
+
+As an auxillary protocol, it is transparent and backwards compatible with the Zcash P2P protocol and consensus layer, and it is also opt-in for wallet usage. Light wallets, as well as full-node wallets, may be able to protect an important aspect of their privacy from hosted service providers with this protocol.
+
+Motivation
+==========
+
+When wallets generate transactions they must transmit them to the Zcash network, and this initial transmission is a weak point for Zcash network privacy: it is possible for network peers to determine which node originated a transaction, even when the P2P protocol is routed over Tor [#]. By associating two different fully shielded transactions with the same originating peer, those transactions are linked by this network metadata, thereby violating a key privacy property that Zeroknowledge proofs protect against on-chain analysis. When a peer does not route its P2P traffic over Tor, this association can also include an IP address, which can reveal some information about the physical location of the wallet.
+
+.. [#] This assertion is an educated bet by the author. FIXME: find published research or supporting evidence to back up this claim.
+
+For mobile light wallets, the originating peer analysis above typically identifies the mobile wallet service provider's peer. Mobile wallets are even more directly vulnerable to their hosting provider in typical deployment architectures, because those hosting providers relay all of a wallet's transactions, and the hosting provider can trivially link those transactions and furthermore typically have more information about the mobile wallet and its user.
+
+This protocol aims to protect the association of which transactions originate with which peers, because each transaction arrives to a random TTRN provider via a new fresh Tor circuit with no history. This may potentially also protect the association of which transactions originate with which mobile light wallets against analysis by mobile wallet service providers.
+
+Design Goals
+============
+
+TODO: refine this section to be more precise/complete.
+
+- Protect linkage between wallets and the transactions they originate.
+- Opt-in for wallets.
+- Feasible for mobile light clients.
+- Auxillary protocol requiring no changes to the Zcash P2P protocol or consensus rules.
+- Incentivize providers via revenue stream.
+- Disincentivize attacker control of a large proportion of the provider network via registration fees.
+
+Non-Goals
+---------
+
+- Protecting the network topological location of providers. (Anyone may determine a transaction first reaches the P2P network from a provider.)
+- Protecting providers from "service identification": anyone may determine a node is a TTRN provider based on its txn emission patterns.
+- Protecting against network attacks which Tor is vulnerable against.
```

Comment:
@defuse brought up some in [their comment](https://github.com/zcash/zips/pull/391#issuecomment-673082232): for example, if your ISP can profile Tor traffic *generally* by bandwidth usage patterns then of course this protocol won't protect against that, since all this protocol does is attempt to use Tor for privacy against other kinds of attacks.